### ISSUE TITLE
openjdk 23 support ?

### ISSUE THREAD

**Issue by a-elkaim** [lucimber/meta-openjdk-temurin#51]
Hi,

Do you plan on adding a recipe for openjdk 23 ?

**Comment by sebveit** [comment]
Hi azsde, since OpenJDK 23 isn't an LTS-version, I'm not going to add it to this layer.

**Comment by a-elkaim** [comment]
Hi,

Understood, thank you for letting me know.

Do you think editing the version values in the 21 recipe would work if I need to create my own recipe ?

**Comment by sebveit** [comment]
It should work, but don't forget to adapt the checksums and maybe the versions of the dependencies (`JVM_RDEPENDS`).

Offtopic: Why do you need OpenJDK 23 instead of 21? You can use an OpenJDK version that is equal to or higher than what your app needs. E.g. you app requires Java 11, than you sould use OpenJDK 11 or 17 of this layer.

**Comment by a-elkaim** [comment]
The project I'm working on has been compiled with JDK 23 (I don't know why, but it seems to be a hard requirement), and as far as I understand, I need a JRE >=23 for it to work.

**Comment by sebveit** [comment]
Not necessarily. It depends on what has been declared on the project as a requirement. The app might only require Java 11 but was compiled with OpenJDK 23 to "byte code" of Java 11. You might check the sources of the app regarding the "hard" version requirement, before you waste time adapting the recipe.

In case you are using Gradle: https://docs.gradle.org/current/userguide/building_java_projects.html#sec:java_cross_compilation

Hope that helps! If you have more questions, just ask.

**Comment by sebveit** [comment]
OpenJDK 25 is available. Closing this issue now.
